ADAL/tests/ADTestURLSessionDataTask.m

```diff
@@ -69,6 +69,11 @@ - (void)resume
         return;
     }
     
+    if (response->_waitSemaphore)
+    {
+        dispatch_semaphore_wait(response->_waitSemaphore, DISPATCH_TIME_FOREVER);
+    }
+    
```

Comment:
No tests have been added using this yet. I'm just added this bit of infra in its own PR so it doesn't clutter up other PRs